File: leetcode-algorithms/050. Pow(x, n)/solution.cpp
class Solution {
public:
    double myPow(double x, int n) {
        long long p;
        if(n<0){
            p = -n;
            x = 1/x;
        }else
            p = n;
        double res = 1.;
        while(p){
            if(p&1){
                res *= x;
            }
            x *= x;
            p >>=1;
        }
        return res;
    }
};

int main()
{
    return 0;
}

File: leetcode-algorithms/050. Pow(x, n)/solution.py
def myPow(x, n):
	"""
	:type x: float
	:type n: int
	:rtype: float
	"""

	# 0 表示正数或0  
	# 1 表示负数
	if n<0:
		x = 1./x
		n = -n
	pow = 1
	while n:
		if n&1:
			pow *= x
		x *= x
		n >>= 1
	return pow



if __name__ == '__main__':
	x = 2.
	y = 10

	res = myPow(x,y)

	print(res)